src/main/scala/co/datamechanics/delight/DelightListener.scala

```diff
@@ -3,31 +3,38 @@ import java.util.concurrent.atomic.AtomicBoolean
 
 import org.apache.spark.internal.Logging
 import org.apache.spark.scheduler._
-import org.apache.spark.{JsonProtocolProxy, SparkConf}
-import org.json4s.jackson.JsonMethods.{compact, render}
+import org.apache.spark.SparkConf
 
 
 class DelightListener(sparkConf: SparkConf) extends SparkListener with Logging {
 
+  /**
+   * Adds Delight version to the Spark config
+   */
+  sparkConf.set("spark.delight.version", Configs.delightVersion)
+
   /**
    * Activates memory metrics collection for Spark 3.0.0 and above.
    * For Spark versions below 3.0.0, these configs have no effect.
    */
   sparkConf.set("spark.executor.processTreeMetrics.enabled", "true")
   sparkConf.set("spark.executor.metrics.pollingInterval", "10ms")
 
+  private val shouldLogDuration = Configs.logDuration(sparkConf)
+
   private val streamingConnector = DelightStreamingConnector.getOrCreate(sparkConf)
 
   /**
   * Conveys whether the logStart event has been sent
   */
   private val logStartEventSent: AtomicBoolean = new AtomicBoolean(false)
 
-  private def logEvent(event: SparkListenerEvent, flush: Boolean = false, blocking: Boolean = false): Unit = {
+  private def logEvent(event: SparkListenerEvent, flush: Boolean = false, blocking: Boolean = false): Unit = Utils.time(
+    shouldLogDuration, "logEvent"
+  ) {
     sendLogStartEventManually()
     try {
-      val eventAsString = compact(render(JsonProtocolProxy.jsonProtocol.sparkEventToJson(event)))
-      streamingConnector.enqueueEvent(eventAsString, flush = flush, blocking = blocking)
+      streamingConnector.enqueueEvent(event, flush = flush, blocking = blocking)
```

Comment:
Why to repeat the names `flush` and `blocking`?

Reply:
Fixed in [`676cc41` (#9)](https://github.com/datamechanics/delight/pull/9/commits/676cc413d1684e7523eaa6c8095bf34b972a6cb9)